app/screens/integration_selector/index.tsx

```diff
@@ -0,0 +1,610 @@
+
+        if (term) {
+            listData = searchResults;
+        }
+
+        if (dataSource === ViewConstants.DATA_SOURCE_USERS) {
+            listData = createProfilesSections(listData as UserProfile[]);
+        }
+
+        if (!dataSource || dataSource === ViewConstants.DATA_SOURCE_DYNAMIC) {
+            listData = (integrationData as DialogOption[]).filter((option) => option.text && option.text.toLowerCase().includes(term));
+        }
+
+        setCustomListData(listData);
+    }, [searchResults, integrationData]);
+
+    useEffect(() => {
+        if (!isMultiselect) {
+            return;
+        }
+
+        setButtons(componentId, {
+            rightButtons: [rightButton],
+        });
+    }, [rightButton, componentId, isMultiselect]);
+
+    useEffect(() => {
+        const multiselectItems: MultiselectSelectedMap = {};
+
+        if (isMultiselect && selected && !([ViewConstants.DATA_SOURCE_USERS, ViewConstants.DATA_SOURCE_CHANNELS].includes(dataSource))) {
+            selected.forEach((opt) => {
+                multiselectItems[opt.value] = opt;
+            });
+
+            setMultiselectSelected(multiselectItems);
+        }
+    }, [multiselectSelected]);
+
+    // Renders
+    const renderLoading = useCallback(() => {
+        if (!loading) {
+            return null;
+        }
+
+        let text;
+        switch (dataSource) {
+            case ViewConstants.DATA_SOURCE_USERS:
+                text = {
+                    id: intl.formatMessage({id: 'mobile.integration_selector.loading_users'}),
+                    defaultMessage: 'Loading Users...',
+                };
+                break;
+            case ViewConstants.DATA_SOURCE_CHANNELS:
+                text = {
+                    id: intl.formatMessage({id: 'mobile.integration_selector.loading_channels'}),
+                    defaultMessage: 'Loading Channels...',
+                };
+                break;
+            default:
+                text = {
+                    id: intl.formatMessage({id: 'mobile.integration_selector.loading_options'}),
+                    defaultMessage: 'Loading Options...',
+                };
+                break;
+        }
+
+        return (
+            <View style={style.loadingContainer}>
+                <FormattedText
+                    {...text}
+                    style={style.loadingText}
+                />
+            </View>
+        );
+    }, [style, dataSource, loading, intl]);
+
+    const renderNoResults = useCallback((): JSX.Element | null => {
+        if (loading || page.current === INITIAL_PAGE) {
+            return null;
+        }
+
+        return (
+            <View style={style.noResultContainer}>
+                <FormattedText
+                    id='mobile.custom_list.no_results'
+                    defaultMessage='No Results'
+                    style={style.noResultText}
+                />
+            </View>
+        );
+    }, [loading, style]);
+
```

Comment:
This should be a callback, but may have performance implications the way it is handled.

Every time you select a channel, the multiselectselected changes, updating this function and ultimately updating every single element in the list.

It is not trivial to solve, so I am ok creating a ticket for this, but it should be solved before ga.